sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -797,34 +821,35 @@ def _handle_features_sbol_to_gb(self, seq_rec: SeqRecord, obj: ComponentGenBankE
                 # else:
 
                 # FIXME: order of features not same as original genbank doc?
-                obj_feat_role = obj_feat.roles[0]
-                # NOTE: The so2gb.csv data file has rows of format 'SO:xxxxxxx,<GenBank_Term>',
-                # and the obj_feat_role returns the URI (i.e 'https://identifiers.org/SO:xxxxxx').
-                # The slicing and subtracting is done to obtain the 'SO:xxxxxxx' portion from the URI.
-                obj_feat_role = obj_feat_role[
-                    obj_feat_role.index(":", 6) - 2:
-                ]
                 # Obtain sequence feature role from Sequence Ontology to GenBank role mappings
+                so_roles = list(filter(None, (tyto_normalize_term(tyto.SO, role) for role in obj_feat.roles)))
                 feat_role = self.DEFAULT_GB_TERM
-                if self.so2gb_map.get(obj_feat_role):
-                    feat_role = self.so2gb_map[obj_feat_role]
+                if len(so_roles):
```

Comment:
This whole block dealing with `so_roles` feels like a good candidate to be extracted into its own function/method. There's enough if/then logic here that it really doesn't need to be inline. I think this is true of a lot of this converter code.